common/plugins/org.jboss.tools.common.jdt.debug/src/org/jboss/tools/common/jdt/debug/tools/ToolsCore.java

```diff
@@ -94,8 +95,32 @@ public Object getMonitoredVM() {
 		return Tools.getInstance().invokeActiveVms(hostname);
 	}
 	
+	/**
+	 * This method was poorly named and is is instead desired to return whether a process
+	 * is able to be monitored or not. 
+	 * 
+	 * @param hostname
+	 * @param vmPid
+	 * @return
+	 * @throws ToolsCoreException
+	 */
+	@Deprecated
 	public static boolean processIsRunning(String hostname, int vmPid) throws ToolsCoreException {
-		return Tools.getInstance().getMonitoredVm(hostname, vmPid) != null;
+		return processIsMonitorable(hostname, vmPid);
+	}
+	
+	public static boolean processIsMonitorable(String hostname, int vmPid) throws ToolsCoreException {
+		// If these throw exceptions, we want them to be rethrown and logged
+		Object host = Tools.getInstance().invokeGetMonitoredHost(hostname);
```

Comment:
Shouldn't we get these two lines  inside try/catch and remove ToolsCoreException from throws?